force-app/main/default/classes/HierarchySettingsService.cls

```diff
@@ -70,11 +70,20 @@ public virtual with sharing class HierarchySettingsService {
         HierarchySettingsModel hierarchySettingsModelToUpdate = this.locateHierarchySettingsMapper()
             .getHierarchySettings();
 
+        if (String.isNotBlank(UTIL_Namespace.getNamespace())) {
+            // replace hierarchy settings field names with prefixed version when in managed context
+            hierarchySettingsChangesVModel = this.addNamespacePrefixToHierarchySettingsChangesFields(
+                hierarchySettingsChangesVModel
+            );
+        }
+
         try {
             // Update Hierarchy Settings fields with single values
             for (
                 String changedSettingName : hierarchySettingsChangesVModel.settingsSingleValueBySettingsName.keySet()
             ) {
+                //changedSettingName = UTIL_Namespace.StrTokenNSPrefix(changedSettingName);
+
                 Schema.DescribeFieldResult fieldDescribeResult = this.locateHierarchySettingsMapper()
                     .getHierarchySettingsField(changedSettingName);
```

Comment:
We can leverage UTIL_Namespace at this juncture without adding a large layer of complexity and using it for the field setting below. Pre-conversion can be a good option, but let's handle namespacing a bit more locally. You would have to use settingsField for the actual put operations below, and replicate that local variable setting, but I feel like the performance difference between those checks is minimal at best with the benefit of reducing complexity.

```suggestion
        try {
            // Update Hierarchy Settings fields with single values
            for (
                String changedSettingName : hierarchySettingsChangesVModel.settingsSingleValueBySettingsName.keySet()
            ) {
                String settingsField = UTIL_Namespace.StrTokenNSPrefix(changedSettingName);

                Schema.DescribeFieldResult fieldDescribeResult = this.locateHierarchySettingsMapper()
                    .getHierarchySettingsField(settingsField);
```